js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx

```diff
@@ -0,0 +1,127 @@
+import {Box, Checkbox, Icon, MenuItem, Suggest} from '@dagster-io/ui-components';
+import isEqual from 'lodash/isEqual';
+import uniqBy from 'lodash/uniqBy';
+import * as React from 'react';
+
+import {AssetGroupSelector} from '../graphql/types';
+import {ClearButton} from '../ui/ClearButton';
+import {buildRepoPathForHuman} from '../workspace/buildRepoAddress';
+
+type Asset = {
+  definition: {
+    groupName: string | null;
+    repository: {name: string; location: {name: string}};
+  } | null;
+};
+
+export function useAssetGroupSelectorsForAssets(assets: Asset[] | undefined) {
+  return React.useMemo(
+    () =>
+      uniqBy(
+        (assets || []).map(buildAssetGroupSelector).filter((a) => !!a) as AssetGroupSelector[],
+        (a) => JSON.stringify(a),
+      ).sort((a, b) => a.groupName.localeCompare(b.groupName)),
+    [assets],
+  );
+}
+
+// We're using a <Suggest /> component because it's convenient to have typeahead behavior,
+// but we want to support mulit-selection. To achieve this, we show checkboxes on the items
+// and override the label text ("2 groups"). Blueprint doesn't need to know what the real
+// selection is, just that one exists.
+const FAKE_SELECTED_ITEM: AssetGroupSelector = {
+  groupName: '-',
+  repositoryLocationName: '-',
+  repositoryName: '-',
+};
+
+export const AssetGroupSuggest: React.FC<{
+  assetGroups: AssetGroupSelector[];
+  value: AssetGroupSelector[];
+  onChange: (g: AssetGroupSelector[]) => void;
+}> = ({assetGroups, value, onChange}) => {
+  const repoContextNeeded = React.useMemo(() => {
+    // This is a bit tricky - the first time we find a groupName it sets the key to `false`.
+    // The second time, it sets the value to `true` + tells use we need to show the repo name
+    const result: {[groupName: string]: boolean} = {};
+    assetGroups.forEach(
+      (group) => (result[group.groupName] = result.hasOwnProperty(group.groupName)),
+    );
+    return result;
+  }, [assetGroups]);
```

Comment:
I've got a similar UI for a Cloud feature, and came to the conclusion that we may as well just always show the code location name, unless perhaps there's only one code location.

Reply:
I like that -- I'll make this show the repo names whenever more than one repo is present. Right now it's cool that it only shows the context when necessary, but the size of the suggestions varies and it's a bit odd. Good call!